File: src/codes/exposed-complexity/py/ksrc.py
class InventoryChecker:
    def check(self, product_id: str) -> bool:
        print(f"在庫確認: {product_id}")
        return True

class PaymentGateway:
    def pay(self, amount: int) -> bool:
        print(f"支払い処理: ¥{amount}")
        return True

class ReceiptPrinter:
    def print(self, product_id: str, amount: int):
        print(f"領収書発行: 商品={product_id}, 金額=¥{amount}")

def complete_purchase(product_id: str, amount: int):
    checker = InventoryChecker()
    payment = PaymentGateway()
    printer = ReceiptPrinter()

    if checker.check(product_id):
        if payment.pay(amount):
            printer.print(product_id, amount)

File: src/codes/exposed-complexity/php/ksrc.php
<?php

class InventoryChecker {
  public function check(string $productId): bool {
    echo "在庫確認: {$productId}\n";
    return true;
  }
}

class PaymentGateway {
  public function pay(int $amount): bool {
    echo "支払い処理: ¥{$amount}\n";
    return true;
  }
}

class ReceiptPrinter {
  public function print(string $productId, int $amount): void {
    echo "領収書発行: 商品={$productId}, 金額=¥{$amount}\n";
  }
}

function completePurchase(string $productId, int $amount): void {
  $checker = new InventoryChecker();
  $payment = new PaymentGateway();
  $printer = new ReceiptPrinter();

  if ($checker->check($productId)) {
    if ($payment->pay($amount)) {
      $printer->print($productId, $amount);
    }
  }
}
